### ISSUE TITLE
Spurious DMP + FSDP warnings for no-parameter modules

### ISSUE THREAD

**Issue by mvpatel2000** [pytorch/pytorch#119832]
### 🐛 Describe the bug

When I recursively wrap modules with FSDP, I get the following warning on modules with no parameters:
```
/usr/lib/python3/dist-packages/torch/distributed/fsdp/_common_utils.py:431: UserWarning: An unexpected prefix is detected. This case  should only happen when using DMP with FSDP. prefix = _fsdp_wrapped_module.transformer.rotary_embedding., submodule_name = _fsdp_wrapped_module
```

I think this should be reproducible by wrapping and training with anything that has no params, eg dropout or rotary embeddings

### Versions

Torch nightly

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @LucasLLC

**Comment by fegin** [comment]
Does the state_dict look good to you?

**Comment by mvpatel2000** [comment]
> Does the state_dict look good to you?

@fegin Yep! I see deterministic elastic resumption. Everything looks fine

**Comment by fegin** [comment]
https://github.com/pytorch/pytorch/pull/126365 should remove the warning.
